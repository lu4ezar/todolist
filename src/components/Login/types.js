// @flow
import type { LoginUserInput } from "../../generated/graphql";

export type Props = {
  values: LoginUserInput,
  isSubmitting: boolean,
  setValues: (LoginUserInput, boolean) => void,
};
