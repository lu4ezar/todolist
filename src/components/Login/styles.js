import { ListItem } from "@material-ui/core";
import styled from "styled-components";

export const StyledForm = styled.form`
  width: 300px;
`;

export const LoginDiv = styled.div`
  color: #2062c5;
  display: flex;
  place-content: center;
  place-items: center;
`;

export const StyledLI = styled(ListItem)`
  &.active {
    background: #94b7e9;
  }
`;
