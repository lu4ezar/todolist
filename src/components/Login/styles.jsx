/* eslint-disable react/jsx-props-no-spreading */
import React from "react";
import { ListItem } from "@mui/material";
import styled from "styled-components";

export const StyledForm = styled.form`
  width: 300px;
`;

export const LoginDiv = styled.div`
  color: #2062c5;
  display: flex;
  place-content: center;
  place-items: center;
`;

// export const StyledLI = styled(ListItem)`
//   &.active {
//     background: #94b7e9;
//   }
// `;

export const StyledLI = styled(
  ({ isDragging, completed, innerRef, ...other }) => (
    <ListItem ref={innerRef} {...other} />
  )
)`
  &.active {
    background: #94b7e9;
  }
`;
