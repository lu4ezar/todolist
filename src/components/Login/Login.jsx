// @flow
import * as React from "react";
import { Formik, Field, ErrorMessage } from "formik";
// import { ExitToApp as LoginIcon } from "@mui/icons-material";
import {
  // IconButton,
  List,
  ListItemText,
  // ClickAwayListener,
  Button,
} from "@mui/material";
import { StyledForm, LoginDiv, StyledLI } from "./styles";
import {
  useCreateUserMutation,
  useLoginMutation,
} from "../../apollo/hooks/user";
import type { Props } from "./types";

const loginOptions = {
  login: "login",
  signup: "signup",
  guest: "login as guest",
};

const initialValues: LoginUserInput = {
  email: "",
  password: "",
};

const Login = ({ values, isSubmitting, setValues }: Props): React.Node => {
  const { email, password } = values;
  const [activeOption, setActiveOption] = React.useState(null);
  const { loginUser, loading } = useLoginMutation();
  const { createUser } = useCreateUserMutation();

  const onSubmit = (e) => {
    e.preventDefault();
    if (activeOption === loginOptions.signup) {
      createUser({ variables: { input: { email, password } } });
    } else {
      loginUser({ variables: { input: { email, password } } });
    }
  };

  if (loading) return "Loading!";

  const handleGuest = () => {
    setActiveOption(loginOptions.guest);
    setValues({ email: "test@mail.com", password: "testtesttest" }, false);
  };
  return (
    <LoginDiv className="open" active>
      <List>
        <StyledLI
          className={activeOption === loginOptions.login ? "active" : ""}
          button
          onClick={() => setActiveOption("login")}
        >
          <ListItemText>LOG IN</ListItemText>
        </StyledLI>
        <StyledLI
          className={activeOption === loginOptions.signup ? "active" : ""}
          button
          onClick={() => setActiveOption("signup")}
        >
          <ListItemText>SIGN UP</ListItemText>
        </StyledLI>
        <StyledLI
          className={activeOption === loginOptions.guest ? "active" : ""}
          button
          onClick={handleGuest}
        >
          <ListItemText>LOGIN AS GUEST</ListItemText>
        </StyledLI>
      </List>
      <StyledForm onSubmit={onSubmit}>
        <Field type="email" name="email" />
        <ErrorMessage name="email" component="div" />
        <Field type="password" name="password" />
        <ErrorMessage name="password" component="div" />
        <Button type="submit" disabled={isSubmitting}>
          Okay
        </Button>
      </StyledForm>
    </LoginDiv>
  );
};

export default (): React.Node => (
  <Formik component={Login} initialValues={initialValues} />
);
