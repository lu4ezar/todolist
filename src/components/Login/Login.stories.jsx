import React from "react";
import Login from "./Login";

export default {
  component: Login,
  title: "Login",
};

function Template() {
  return <Login />
}

export const Default = Template.bind({});
Default.args = {};

export const Enabled = Template.bind({});
Enabled.args = {};
