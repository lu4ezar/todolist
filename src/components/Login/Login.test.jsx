import React from "react";
import { render } from "../../utils/test-utils";

import Login from ".";

describe("Login", () => {
  it("should render without error", async () => {
    const { container } = render(<Login />);
    expect(container).toMatchSnapshot();
  });
});
