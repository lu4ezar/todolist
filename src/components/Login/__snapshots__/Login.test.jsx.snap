// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render without error 1`] = `
<div>
  <div>
    <button
      aria-label="Login"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10.09 15.59L11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5c-1.11 0-2 .9-2 2v4h2V5h14v14H5v-4H3v4c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <div
      class="sc-gsTCUz hHKidQ open"
    >
      <ul
        class="MuiList-root MuiList-padding"
      >
        <div
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root sc-dlfnbm ApxIJ MuiListItem-gutters MuiListItem-button"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              LOG IN
            </span>
          </div>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
        <div
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root sc-dlfnbm ApxIJ MuiListItem-gutters MuiListItem-button"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              SIGN UP
            </span>
          </div>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
        <div
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root sc-dlfnbm ApxIJ MuiListItem-gutters MuiListItem-button"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              LOGIN AS GUEST
            </span>
          </div>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
      </ul>
    </div>
  </div>
</div>
`;
